.github/workflows/tests.yml

```diff
@@ -102,17 +102,50 @@ jobs:
         if: matrix.target == 'plugins'
         run: bin/rake plugin:install_all_official
 
-      - name: Create database
+      - name: Fetch app state cache
+        uses: actions/cache@v2
+        id: app-cache
+        with:
+          path: tmp/app-cache
+          key: >- # postgres version, hash of migrations, "parallel?"
+            ${{ runner.os }}-
+            ${{ hashFiles('.github/workflows/*') }}-
+            ${{ matrix.postgres }}-
+            ${{ hashFiles('db/**/*') }}-
+            ${{ hashFiles('plugins/**/db/**/*') }}-
+            ${{ matrix.build_type=='backend' && matrix.target=='core' }}
```

Comment:
Trying to wrap my head around it. 😃 How many versions of the cache will this make?

1. The last part (is it still necessary?) splits the key space into "core-backend" (`true`) and "the rest"(`false`).
2. "The rest" are split into two again by `hashFiles('plugins/**/db/**/*')`: one hash for `target == plugins` (since it has more files in those paths) and another one for `target == core`.

So… `-core-true`, `-core-false`, `-plugins-false` (`-plugins-true` doesn't exist, coz target can't be both core and plugins) That gives us 3 versions of the cache? I think?

Reply:
Bottom line: I think we don't need that last line. 🙂

Reply:
> So… `-core-true`, `-core-false`, `-plugins-false` (`-plugins-true` doesn't exist)

Yes, but `-core-true` needs to be split in half again. The core frontend tests only use one database, while the core backend tests use multiple database. That's why we need the last line.

Reply:
Updated to make the key more readable